src/filterA.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @since TODO
+ */
+import { HKT, Kind, Kind2, Kind3, Kind4, URIS, URIS2, URIS3, URIS4 } from 'fp-ts/lib/HKT'
+import { Applicative, Applicative1, Applicative2, Applicative3, Applicative4 } from 'fp-ts/lib/Applicative'
+import * as A from 'fp-ts/lib/Array'
+import * as O from 'fp-ts/lib/Option'
+
+/**
+ * TODO describe here
+ *
+ * @example
+ * import { io, IO } from 'fp-ts/lib/IO'
+ * import { filterA } from 'fp-ts-contrib/lib/filterA'
+ *
+ * const filterAIO = filterA(io)
+ *
+ * const p = (n: number): IO<boolean> => io.of(n % 2 === 0)
+ *
+ * filterAIO(p)([1, 2, 3, 4, 5])() // [2, 4]
```

Comment:
you can write an assert here

```ts
assert.deepStrictEqual(filterAIO(p)([1, 2, 3, 4, 5])(), [2, 4])
```

will be type checked / executed while generating the docs